src/flamenco/runtime/program/fd_bpf_loader_v3_program.c

```diff
@@ -1038,12 +1042,19 @@ process_loader_upgradeable_instruction( fd_exec_instr_ctx_t * instr_ctx ) {
 
     /* https://github.com/anza-xyz/agave/blob/574bae8fefc0ed256b55340b9d87b7689bcdf222/programs/bpf_loader/src/lib.rs#L825-L845 */
     /* Load and verify the program bits */
-    ulong programdata_size = fd_ulong_sat_add( SIZE_OF_PROGRAM, programdata_len );
     if( FD_UNLIKELY( buffer_data_offset>buffer->meta->dlen ) ) {
       FD_LOG_WARNING(( "Buffer data offset is out of bounds" ));
       return FD_EXECUTOR_INSTR_ERR_ACC_DATA_TOO_SMALL;
     }
-    err = deploy_program( instr_ctx, buffer->data+buffer_data_offset, programdata_size );
+
+    uchar * programdata_data = buffer->data    + buffer_data_offset;
+    ulong   programdata_size = programdata_len - buffer_data_offset;
+
+    /* We use programdata_size instead of UpgradeableLoaderState::size_of_program().saturating_add(programdata_len)
+       because that is the length of our buffer, and that argument to deploy_program (account_size) in the rust code is not used
+       in our version of deploy_program because we do not have a program cache
+       https://github.com/anza-xyz/agave/blob/da6f7f2d317ed978701b08c3cc3e8d62756e565a/programs/bpf_loader/src/lib.rs#L146 */
+    err = deploy_program( instr_ctx, programdata_data, programdata_size );
```

Comment:
```suggestion
    err = deploy_program( instr_ctx, buffer_data, buffer_data_len );
```

Reply:
This is equivalent to this in Agave:  https://github.com/anza-xyz/agave/blob/3ee204b5cba93211fa0e07dc9238372eeb7933db/programs/bpf_loader/src/lib.rs#L839